### PR TITLE
PYTHON-3943 Serverless failure - TestUnifiedRunCursorCommand

### PR DESCRIPTION
**Pull request by NoahStapp**
This issue is caused by a side-effect of `pymongo.message._op_msg`. Because we pass `_GetMore._as_command` as the `command` parameter, this line directly adds the `$db` value to ` _GetMore._as_command`: https://github.com/mongodb/mongo-python-driver/blob/14640256cb0e4be21a94dbd3926b490ad8257bbc/pymongo/message.py#L746

As a result, when the final `getMore` of the test hits this line: https://github.com/mongodb/mongo-python-driver/blob/14640256cb0e4be21a94dbd3926b490ad8257bbc/pymongo/server.py#L127 
it does not get `$db` added to its command spec for the test listener, causing the test to fail.

### CODE REVIEW

File: pymongo/server.py [hunk 1]
```diff
@@ -132,6 +132,8 @@ def run_operation(
 
         if publish:
             cmd, dbn = operation.as_command(conn)
+            if "$db" not in cmd and dbn is not None:
```
Comment (ShaneHarvey) [inline pymongo/server.py:135]: 2 questions: Is dbn ever None here? Could we move this logic to as_command?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Is it possible to add a test for this that runs on regular servers?

File: pymongo/server.py (hunk 1 above)
Comment (ShaneHarvey) [reply pymongo/server.py:135]: I still have this question.

**Comment by ShaneHarvey** [comment]
> Is it possible to add a test for this that runs on regular servers?

This seems possible. Just add a test that checks for $db on OP_MSG an exhaust cursor.

File: pymongo/server.py (hunk 1 above)
Comment (NoahStapp) [reply pymongo/server.py:135]: `dbn` does not appear to ever be `None`, but moving this logic to `as_command` does not work, the test fails with the same "missing $db" error message.

**Comment by NoahStapp** [comment]
> > Is it possible to add a test for this that runs on regular servers?
> 
> This seems possible by adding a test that checks for $db on an OP_MSG exhaust cursor.

This test would verify that `$db` is set on exhaust cursors?

**Comment by ShaneHarvey** [comment]
Yes, using the CommandListener to assert on the CommandStartedEvents for find/getMore commands.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Nice test. I have some comments that would help make it more robust.

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**